### PR TITLE
fixing the position and display of the segment labels during sketch mode

### PR DESCRIPTION
**Pull request by nadr0**
closes https://github.com/KittyCAD/modeling-app/issues/3076

The position of the label grew with the scale since it is in the offsetFromMidPoint. We need the parent container to be position properly and place the label inside of it, we don't need to reposition the label within the parent container.

- Updated the styling to get closer to our designs
- Support both light and dark mode
- Added background color to have separate between the overlaid text and the line
- Positioned the text at the midpoint of the line and have the text be centered
- Works at different zoom levels

image
image

### CODE REVIEW

**Comment by franknoirot** [comment]
Might need some padding around the text, on dark mode it looks a little accidental. `py-0.5 px-2` maybe? You might just need `pt-0.5`, the line-height of that font messes with the visual padding

**Comment by lf94** [comment]
1. can we add a bit of padding
2. can we remove the units (already globally specified)
3. maybe add the smallest rounded corners
4. (stretch goal) can we remove the sketch coloring? why doesnt lightmode have an internal color?

**Comment by franknoirot** [comment]
`4` is the engine-side scene, which is dimmed when in sketch mode and frozen. I don't know why it's not visible in the light mode screenshot, probably because it was the first taken and there was not yet an engine-side Solid2D to render

**Comment by nadr0** [comment]
- [x] can we add a bit of padding
- [x] can we remove the units (already globally specified)
- [x] maybe add the smallest rounded corners
- [x] stretch goal) can we remove the sketch coloring? why doesnt lightmode have an internal color?

Frank is right about the line height padding, it can be a little odd with this font. 

image
image

**Comment by lf94** [comment]
question: how does this look with a border color matching line colors?

**Comment by lf94** [comment]
I personally like it because it kind of ties together the line through the border - @franknoirot @Irev-Dev ?

**Comment by Irev-Dev** [comment]
I'll let Frank make the call on this.

**Comment by nadr0** [comment]
Suggestion from Max
>in the drawing world you would align the text angle with the direction of the line

Added aligning the text to the slope of the line. 

image
image


https://github.com/user-attachments/assets/904e644d-c4d4-4dd2-b370-084e11b6e787

**Comment by franknoirot** [comment]
Love it. I don't love the border around because it makes it feel like an input to me, I think that might increase the user complaints around the lack of ability to do the "double-click-to-edit-length" user flow. But I think this is awesome and I want it in more. I say punch it

**Comment by lf94** [comment]
>  I don't love the border around because it makes it feel like an input to me, I think that might increase the user complaints around the lack of ability to do the "double-click-to-edit-length" user flow.

Isn't that a benefit? That users _can_ do this in some upcoming PR?

File: src/index.css [hunk 1]
```diff
@@ -267,15 +267,16 @@ code {
 }
 
 .segment-length-label-text {
-  transform: translate(var(--x, 0), var(--y, 0));
+  transform: translate(var(--x, 0), var(--y, 0)) rotate(var(--degree, 0));
```
Comment (franknoirot) [inline src/index.css:270]: So clean!

**Review by franknoirot — APPROVED** [review]
Nice job!

**Comment by Irev-Dev** [comment]
Sorry this is late, and I don't think it's a huge problem since there's an obvious work around for users "zoom in"

But did notice at scale it disappears too late and covers the entire segment

image
```ts
const sketch001 = startSketchOn('XZ')
  |> startProfileAt([3.44, 3.48], %)
  |> line([1500.31, 1.06], %)
```

**Comment by lf94** [comment]
Yeah the follow up PR could be when two boxes overlap they both disappear.

**Comment by Irev-Dev** [comment]
Sorry noticing all the things post merge.

https://github.com/user-attachments/assets/8e2b28a9-7163-4e00-91ed-f8487bb90c2c

It's kinda charming tbh.

**Pull request merged**